pac_windows.go

```diff
@@ -66,10 +39,20 @@ func getProxyForURL(pacfileURL, URL string) (string, error) {
 
 	getProxyForUrl := winHttp.NewProc("WinHttpGetProxyForUrl")
 
+	dwFlags := fWINHTTP_AUTOPROXY_CONFIG_URL
+	dwAutoDetectFlags := autoDetectFlag(0)
+	pfURLptr := pacfileURLPtr
+
+	if autoDetect {
+		dwFlags = fWINHTTP_AUTOPROXY_AUTO_DETECT
+		dwAutoDetectFlags = fWINHTTP_AUTO_DETECT_TYPE_DNS_A | fWINHTTP_AUTO_DETECT_TYPE_DHCP
+		pfURLptr = nil
+	}
+
 	options := tWINHTTP_AUTOPROXY_OPTIONS{
-		dwFlags:                fWINHTTP_AUTOPROXY_CONFIG_URL, // adding cache might cause issues: https://github.com/mattn/go-ieproxy/issues/6
-		dwAutoDetectFlags:      0,
-		lpszAutoConfigUrl:      pacfileURLPtr,
+		dwFlags:                dwFlags, // adding cache might cause issues: https://github.com/mattn/go-ieproxy/issues/6
+		dwAutoDetectFlags:      dwAutoDetectFlags,
+		lpszAutoConfigUrl:      pfURLptr,
 		lpvReserved:            nil,
 		dwReserved:             0,
 		fAutoLogonIfChallenged: true, // may not be optimal https://msdn.microsoft.com/en-us/library/windows/desktop/aa383153(v=vs.85).aspx
```

Comment:
the lpsz* fielsd of info all need to be freed after we have finish reading them. to do so, we need a function which we pass them to, that works as follows.  If they are non-nil, it passes them to GlobalFree in teh Kernel32 dll, using the usual uintptr(unsafe.Pointer(lpsz*goesHere)) construct.

For lpszProxy, we must get its value with StringFromUTF16Ptr first;.  For lpszProxyByPass (which we don't read) we should still pass it to our free function.

BTW, the lack of freeing is not strictly related to this PR, since the issue predates this PR. but we may as well fix it now.